lib/cli/src/commands/binfmt.rs

```diff
@@ -0,0 +1,153 @@
+use anyhow::{Context, Result};
+use std::env;
+use std::fs;
+use std::io::Write;
+use std::os::unix::ffi::OsStrExt;
+use std::os::unix::fs::MetadataExt;
+use std::path::{Path, PathBuf};
+use structopt::StructOpt;
+use Action::*;
+
+#[derive(StructOpt, Clone, Copy)]
+enum Action {
+    /// Register wasmer as binfmt interpreter
+    Register,
+    /// Unregister a binfmt interpreter for wasm32
+    Unregister,
+    /// Soft unregister, and register
+    Reregister,
+}
+
+/// Unregister and/or register wasmer as binfmt interpreter
+///
+/// Check the wasmer repository for a systemd service definition example
+/// to automate the process at start-up.
+#[derive(StructOpt)]
+pub struct Binfmt {
+    // Might be better to traverse the mount list
+    /// Mount point of binfmt_misc fs
+    #[structopt(long, default_value = "/proc/sys/fs/binfmt_misc/")]
+    binfmt_misc: PathBuf,
+
+    #[structopt(subcommand)]
+    action: Action,
+}
+
+// Quick safety check:
+// This folder isn't world writeable (or else its sticky bit is set), and neither are its parents.
+//
+// If somebody mounted /tmp wrong, this might result in a TOCTOU problem.
+fn seccheck(path: &Path) -> Result<()> {
+    if let Some(parent) = path.parent() {
+        seccheck(parent)?;
+    }
+    let m = std::fs::metadata(path)
+        .with_context(|| format!("Can't check permissions of {}", path.to_string_lossy()))?;
+    anyhow::ensure!(
+        m.mode() & 0o2 == 0 || m.mode() & 0o1000 != 0,
+        "{} is world writeable and not sticky",
+        path.to_string_lossy()
+    );
+    Ok(())
+}
+
+impl Binfmt {
+    /// execute [Binfmt]
+    pub fn execute(&self) -> Result<()> {
+        if !self.binfmt_misc.exists() {
+            panic!("{} does not exist", self.binfmt_misc.to_string_lossy());
+        }
+        let temp_dir;
+        let specs = match self.action {
+            Register | Reregister => {
+                temp_dir = tempfile::tempdir().context("Make temporary directory")?;
+                seccheck(temp_dir.path())?;
+                let bin_path_orig: PathBuf = env::args_os()
+                    .nth(0)
+                    .map(Into::into)
+                    .filter(|p: &PathBuf| p.exists())
+                    .context("Cannot get path to wasmer executable")?;
+                let bin_path = temp_dir.path().join("wasmer-binfmt-interpreter");
+                fs::copy(&bin_path_orig, &bin_path).context("Copy wasmer binary to temp folder")?;
```

Comment:
I don't understand why the specialized binfmt version of wasmer is copied to temp. The binfmt magic will not survive a reboot like that. why not just create a symlink of wasmer to the new name in the same folder?

Reply:
The binfmt registration is kernel side and does not survive a reboot anyway. (And the file in `/tmp` is deleted once `temp_dir` is dropped. This is allowed because of the `F` flag in `...:PFC`, which also makes sure the interpreter works across mount namespaces / chroots / ...) There needs to be some kind of systemd/openrc/anacron/... service that re-does the registration on every boot. I can add that to the installer script if you want, but it doesn't fit well with the current "install to user's home folder".

Reply:
I see.
I guess we can provide hint in doc to do an actual system installation, with binfmt script to be used by systemd binfmt deamon as an example.